main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
Given that syslog is a 1-way protocol, how would syslog guarantee it's using a charset that the destination understands?  Particularly in the case of remote logging.

When in doubt, I fall back on a mentor's advice, in this case, [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle):

> Be conservative in what you do, be liberal in what you accept from others

Of course, Jon wrote that well before zero-day exploits were a common phenomenon, if I had to recoin his advice, I'd write it:

> Be conservative in what you do, and extremely paranoid in what you accept from others